flow/connectors/mysql/validate.go

```diff
@@ -117,10 +117,36 @@ func (c *MySqlConnector) CheckBinlogSettings(ctx context.Context, requireRowMeta
 		}
 	}
 
-	binlogRowImage := shared.UnsafeFastReadOnlyBytesToString(row[2].AsString())
-	if binlogRowImage != "FULL" {
-		c.logger.Warn("binlog_row_image should be set to 'FULL' to avoid missing data",
-			slog.String("binlog_row_image", strings.Clone(binlogRowImage)))
+	if checkRowValueOptions {
+		binlogRowValueOptions := shared.UnsafeFastReadOnlyBytesToString(row[4].AsString())
+		if binlogRowValueOptions != "" {
+			return errors.New("binlog_row_value_options must be disabled, currently " + binlogRowValueOptions)
+		}
+	}
+
+	// AWS RDS/Aurora has its own binlog retention setting that we need to check, minimum 24h
+	// check RDS/Aurora binlog retention setting
+	if rs, err := c.Execute(ctx, "SELECT value FROM mysql.rds_configuration WHERE name='binlog retention hours'"); err != nil {
+		var mErr *mysql.MyError
+		if errors.As(err, &mErr) && mErr.Code == mysql.ER_NO_SUCH_TABLE {
+			// Table doesn't exist, which means this is not RDS/Aurora
+			slog.Warn("mysql.rds_configuration table does not exist, skipping Aurora/RDS binlog retention check")
+			return nil
+		}
+		return errors.New("failed to check RDS/Aurora binlog retention hours: " + err.Error())
+	} else if len(rs.Values) > 0 {
+		binlogRetentionHoursStr := shared.UnsafeFastReadOnlyBytesToString(rs.Values[0][0].AsString())
+		if binlogRetentionHoursStr == "" {
+			return errors.New("RDS/Aurora setting 'binlog retention hours' should be atleast 24, currently unset")
+		}
+		slog.Info("binlog retention hours", "binlogRetentionHours", binlogRetentionHoursStr)
+		if binlogRetentionHours, err := strconv.Atoi(binlogRetentionHoursStr); err != nil {
+			return errors.New("failed to parse RDS/Aurora setting 'binlog retention hours': " + err.Error())
+		} else if binlogRetentionHours < 24 {
+			return errors.New("RDS/Aurora setting 'binlog retention hours' should be at least 24, currently " + binlogRetentionHoursStr)
+		}
+	} else {
+		slog.Warn("binlog retention hours returned nothing, skipping Aurora/RDS binlog retention check")
```

Comment:
I think this scenario means 0h retention

Reply:
in that case the query returns NULL which maps to empty string L139-L141